app/src/main/java/org/simple/clinic/teleconsultlog/drugduration/DrugDurationUiActions.kt

```diff
@@ -0,0 +1,5 @@
+package org.simple.clinic.teleconsultlog.drugduration
+
+interface DrugDurationUiActions {
+  fun showBlankDurationError()
```

Comment:
This should be done in the UI renderer, right?

We have kept these in the UI actions as effects when migrating from Mobius because it's simpler that way. But for new screens, we should be tracking any error states to display in the model and then render them in the UI renderer.

An exception is when error messages are shown as modals.

Reply:
I have question regarding this approach. In this case doesn't UI actions makes more sense? because the error message visibility not only is dependent on the duration validation result but also on the duration text changes. How does that translate to the model 🤔

Reply:
Basically anything that is renderered as part of the screen should be done through the model. What this means is that we have a nullable property in the model, say a `ValidationResult` enum with `[NOT_VALIDATED, BLANK]` states and then change the state in the model based on what we want to render.

Reply:
Ah okay. Just to confirm that means whenever the duration text changes we need to update the model twice, once for updating the duration in the model and second one is to update the validation result to set it to `NOT_VALIDATED` in order to remove the error whenever the duration text changes. WDYT?

Reply:
Something like this
```kt
      is DurationChanged -> next(
          model
              .durationChanged(event.duration)
              .validationResultChanged(NOT_VALIDATED)
      )
```

Reply:
This works, but I prefer to make it so that `durationChanged` could internally set the validation result to `NOT_VALIDATED`. We could then add another method that is `.invalid(ValidationResult)` which can be used when the model is invalid.

Reply:
Yeah, I was thinking about doing that but felt like it's a hidden behaviour 😅

Reply:
It's not really hidden, the business logic is part of the model.